Sources/Common/ProgressiveRouteType.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  ProgressiveRouteType.swift
+//  OperaSwift
+//
+//  Created by Diego Ernst on 5/11/17.
+//
+//
+
+import Alamofire
+import Foundation
+
+public protocol ProgressiveRouteType: AdaptedRouteType {
```

Comment:
I'm thinking if you can just get rid of `AdaptedRouteType` and implement all its stuff in here